auth/state.go

```diff
@@ -61,27 +66,34 @@ type GormOauthStateReferenceRepository struct {
 
 // Delete deletes the reference with the given id
 // returns NotFoundError or InternalError
-func (r *GormOauthStateReferenceRepository) Delete(ctx context.Context, ID uuid.UUID) error {
-	if ID == uuid.Nil {
+func (r *GormOauthStateReferenceRepository) Delete(ctx context.Context, state string) error {
+	if &state == nil {
 		log.Error(ctx, map[string]interface{}{
-			"oauth_state_reference_id": ID.String(),
-		}, "unable to find the oauth state reference by ID")
-		return errors.NewNotFoundError("oauth state reference", ID.String())
+			"oauth_state_reference_state": state,
+		}, "unable to find the oauth state reference by state")
+		return errors.NewNotFoundError("oauth state reference", state)
+	}
+
+	reference, err := r.Load(ctx, state)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"state": state,
+		}, "Could not find oauth state reference by state")
+		return errors.NewNotFoundError("oauth state reference", state)
 	}
-	reference := OauthStateReference{ID: ID}
 	tx := r.db.Delete(reference)
 
 	if err := tx.Error; err != nil {
 		log.Error(ctx, map[string]interface{}{
-			"oauth_state_reference_id": ID.String(),
+			"oauth_state_reference_state": state,
 		}, "unable to delete the oauth state reference")
 		return errors.NewInternalError(ctx, err)
 	}
 	if tx.RowsAffected == 0 {
 		log.Error(ctx, map[string]interface{}{
-			"oauth state reference": ID.String(),
+			"oauth state reference": state,
```

Comment:
@kishansagathiya could you take care of this review comment ?
for the key in the error log message - Use smaller case letters with words joined by "_" , example, "state_id" instead of "State ID",